packages/back-end/src/controllers/stripe.ts

```diff
@@ -1,144 +1,168 @@
-const stripe = new Stripe(STRIPE_SECRET || "", { apiVersion: "2020-08-27" });
-
-async function updateSubscription(subscription: string | Stripe.Subscription) {
-  // Make sure we have the full subscription object
-  if (typeof subscription === "string") {
-    subscription = await stripe.subscriptions.retrieve(subscription);
-  }
-
-  const stripeCustomerId =
-    typeof subscription.customer === "string"
-      ? subscription.customer
-      : subscription.customer.id;
-
-  await updateOrganizationByStripeId(stripeCustomerId, {
-    subscription: {
-      id: subscription.id,
-      qty: subscription.items.data[0].quantity || 1,
-      trialEnd: subscription.trial_end
-        ? new Date(subscription.trial_end * 1000)
-        : null,
-      status: subscription.status,
-    },
-  });
-}
-export async function postStartTrial(
-  req: AuthRequest<{ qty: number; name: string }>,
+  getNumberOfMembersAndInvites,
+  getOrgFromReq,
+} from "../services/organizations";
+import {
+  updateSubscriptionInDb,
+  stripe,
+  getCoupon,
+  getPrice,
+  getStripeCustomerId,
+} from "../services/stripe";
+import { SubscriptionQuote } from "../../types/organization";
+
+export async function postNewSubscription(
+  req: AuthRequest<{ qty: number }>,
   res: Response
 ) {
-  const { qty, name } = req.body;
+  const { qty } = req.body;
 
-  // If user already has a subscription, return immediately
-  if (req.organization?.subscription?.id) {
-    return res.status(200).json({
-      status: 200,
-    });
+  req.checkPermissions("organizationSettings");
+
+  const { org } = getOrgFromReq(req);
+
+  const desiredQty = getNumberOfMembersAndInvites(org);
+
+  if (desiredQty !== qty) {
+    throw new Error(
+      "Number of users is out of date. Please refresh the page and try again."
+    );
   }
 
-  try {
-    // Create organization first if needed
-    if (!req.organization) {
-      if (name.length < 3) {
-        throw new Error("Company name must be at least 3 characters long");
-      }
-      req.organization = await createOrganization(
-        req.email || "",
-        req.userId || "",
-        name,
-        ""
-      );
-    }
+  const stripeCustomerId = await getStripeCustomerId(org);
 
-    // Create customer in Stripe if not exists
-    if (!req.organization.stripeCustomerId) {
-      const resp = await stripe.customers.create({
-        email: req.email || "",
-        name: req.name || "",
-        metadata: {
-          user: req.userId || "",
-          organization: req.organization.id,
-        },
-      });
-      req.organization.stripeCustomerId = resp.id;
-
-      await updateOrganization(req.organization.id, {
-        stripeCustomerId: resp.id,
-      });
+  const existingSubscriptions = await stripe.subscriptions.list({
+    customer: stripeCustomerId,
+  });
+
+  existingSubscriptions.data.forEach((subscription) => {
+    if (subscription.status === ("active" || "past_due" || "trialing")) {
```

Comment:
@jdorn I added checks for 3 potential "active-ish" statuses - let me know if you like that or if you'd only like to throw the error when it's "active".